generators/client/templates/angular/webpack/proxy.conf.js.ejs

```diff
@@ -17,7 +17,7 @@
  limitations under the License.
 -%>
 function setupProxy() {
```

Comment:
Parsing argv is a bit hack, instead of exporting `setupProxy()` export `setupProxy` itself and check if options/targetOptions are available.
```suggestion
function setupProxy(options) {
```

Reply:
I have tried the same. but the 'config'  or 'option' is not available in the proxy conf file getting undefined. I will check any other possibility

Reply:
Try this.
```
diff --git a/angular.json b/angular.json
index 52723dc..5c89a37 100644
--- a/angular.json
+++ b/angular.json
@@ -85,7 +85,6 @@
           "builder": "@angular-builders/custom-webpack:dev-server",
           "options": {
             "browserTarget": "angular:build:development",
-            "proxyConfig": "./webpack/proxy.conf.js",
             "port": 4200
           },
           "configurations": {
diff --git a/webpack/proxy.conf.js b/webpack/proxy.conf.js
index 1dbfe42..318ac1b 100644
--- a/webpack/proxy.conf.js
+++ b/webpack/proxy.conf.js
@@ -1,5 +1,4 @@
-function setupProxy() {
-  const tls = process.env.TLS;
+function setupProxy({ tls }) {
   const conf = [
     {
       context: [
@@ -21,4 +20,4 @@ function setupProxy() {
   return conf;
 }

-module.exports = setupProxy();
+module.exports = setupProxy;
diff --git a/webpack/webpack.custom.js b/webpack/webpack.custom.js
index ff7e774..9080e63 100644
--- a/webpack/webpack.custom.js
+++ b/webpack/webpack.custom.js
@@ -8,6 +8,7 @@ const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPl
 const WebpackNotifierPlugin = require('webpack-notifier');
 const CopyWebpackPlugin = require('copy-webpack-plugin');
 const ESLintPlugin = require('eslint-webpack-plugin');
+const proxyConfig = require('./proxy.conf');

 const environment = require('./environment');

@@ -49,6 +50,9 @@ module.exports = async (config, options, targetOptions) => {
     );
   }
   if (targetOptions.target === 'serve' || config.watch) {
+    if (config.devServer) {
+      config.devServer.proxy = proxyConfig({ tls });
+    }
     config.plugins.push(
       new BrowserSyncPlugin(
         {

```

Reply:
Working perfectly as expected. Thank you @mshima. I have committed the latest changes as suggested. 
I have moved proxy creation outside. because https://github.com/jhipster/generator-jhipster/blob/main/generators/client/templates/angular/webpack/webpack.custom.js.ejs#L73 since we need to add proxy for microservice application